src/Middleware/AuthenticationMiddleware.php

```diff
@@ -118,21 +123,25 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
         $request = $request->withAttribute('authentication', $service);
         $request = $request->withAttribute('authenticationResult', $result['result']);
 
-        $response = $result['response'];
-
         try {
-            return $next($request, $response);
+            $response = $handler->handle($request);
+            $authenticator = $service->getAuthenticationProvider();
+
+            if ($authenticator !== null && !$authenticator instanceof StatelessInterface) {
+                $return = $service->persistIdentity($request, $response, $result['result']->getData());
```

Comment:
This will trip people up. The session will still have the user data in it, but it will always be from the previous request and not the current one.

Reply:
It will only trip those who incorrectly use session directly to get identity instead of the identity attribute of request instance.

The alternative is to have two separate methods for persisting identity. One that is called with request object before calling `$handler->handler()` and one that's called after with response object.

Reply:
Him adding more methods does increase complexity. Perhaps we can rely on docs showing how to get the identity correctly and worry about this if that doesn't work. :)

Reply:
I have updated the "quick start" page warning against directly using session.